Makefile

```diff
@@ -1,30 +1,103 @@
-export CABAL := $(shell cabal update 2>&1 >/dev/null && cabal list-bin -v0 --project-dir libraries/Cabal cabal-install:exe:cabal)
+SHELL := bash
+.ONESHELL:
+.SHELLFLAGS := -eu -o pipefail -c
+.DELETE_ON_ERROR:
+MAKEFLAGS += --warn-undefined-variables
+MAKEFLAGS += --no-builtin-rules
 
-CPUS=$(shell mk/detect-cpu-count.sh)
+CABAL0 ?= cabal
+CABAL  ?= _stage0/bin/cabal
+GHC    ?= ghc
 
-# Use CPU cores + 1 if not already set
-THREADS=${THREADS:-$((CPUS + 1))}
+define LIB =
+BOLD_RED='\033[1;31m'
+RESET='\033[0m'
 
-all: $(CABAL) ./booted
-	GHC=ghc-9.8.4 ./Build.hs
+log() {
+  echo -e "$${BOLD_RED}+$${RESET} $$@" >&2 
+  "$$@" 
+}
+endef
 
-cabal: $(CABAL)
+%/configure : %/configure.ac
+	@$(LIB)
+	log autoreconf $(@D)
 
-$(CABAL):
-	cabal build --project-dir libraries/Cabal cabal-install:exe:cabal
+CABAL_FLAGS += --store-dir $(OUT)/store --logs-dir $(OUT)/logs
 
-./booted:
-	./boot
-	touch $@
+CABAL_INSTALL_FLAGS += --builddir $(OUT)/build
+CABAL_INSTALL_FLAGS += --with-compiler $(GHC)
+CABAL_INSTALL_FLAGS += --installdir $(OUT)/bin
+CABAL_INSTALL_FLAGS += --overwrite-policy=always
+# If we copy the executables then ghc will recognise _stage1 as topdir (rather than a path in the store)
+CABAL_INSTALL_FLAGS += --install-method=copy
```

Comment:
We should not build `cabal` as part of this build. We should assume `cabal` as a build tool to just exist. We specifically want to get rid of this symbiotic GHC / hadrian setting.

The nix shell does contain a properly patched cabal. If we didn't we should build it somewhere else and grab binary blobs.

Reply:
We need to build cabal because we have modified it.

Reply:
Yes, but I _really_, _really_ do not want the build system to be symbiotic with the source. We can for now just assume we must bring a modified cabal. We could even have a configure check to ensure the cabal is acceptable. But we must not build the build tool as part of the build script. We don't build gcc, a bootstrap ghc, ... either.